modules/core/src/test/java/org/apache/ignite/internal/commandline/CommandHandlerParsingTest.java

```diff
@@ -277,6 +291,59 @@ public void testParseAndValidateSSLArguments() {
         }
     }
 
+    /**
+     * Tests parsing user attribute arguments from string.
+     */
+    @Test
+    public void testParseUserAttrStringArguments() {
+        List<String> cmdArgs = new ArrayList<>(DFLT_ARGS);
+
+        cmdArgs.add(CommandList.TX.text());
+        cmdArgs.add(CommonArgParser.CMD_USER_ATTR);
+        cmdArgs.add("attr1=val1,attr2=val2,attr3=mode=test");
+
+        ConnectionAndSslParameters args = parseArgs(cmdArgs);
+
+        assertEquals(args.userAttributes().get("attr1"), "val1");
+        assertNotEquals(args.userAttributes().get("attr1"), "val2");
+        assertEquals(args.userAttributes().get("attr3"), "mode=test");
+    }
+
+    /**
+     * Tests parsing user attribute arguments from file.
+     */
+    @Test
+    public void testParseUserAttrFileArguments() {
+        List<String> cmdArgs = new ArrayList<>(DFLT_ARGS);
+
+        cmdArgs.add(CommandList.TX.text());
+        cmdArgs.add(CommonArgParser.CMD_USER_ATTR_PATH);
+        cmdArgs.add(IgniteUtils.resolveIgnitePath(USER_ATTRIBUTE_PATH).getAbsolutePath());
+
+        ConnectionAndSslParameters args = parseArgs(cmdArgs);
+
+        assertEquals(args.userAttributes().get("attr3"), "val3");
+        assertEquals(args.userAttributes().get("attr4"), "mode=test");
+    }
+
+    /**
+     * Tests parsing user attribute arguments from file.
+     */
+    @Test
+    public void testParseUserAttrStringFileArguments() {
+        List<String> cmdArgs = new ArrayList<>(DFLT_ARGS);
+
+        cmdArgs.add(CommandList.TX.text());
+        cmdArgs.add(CommonArgParser.CMD_USER_ATTR_PATH);
+        cmdArgs.add(IgniteUtils.resolveIgnitePath(USER_ATTRIBUTE_PATH).getAbsolutePath());
+        cmdArgs.add(CommonArgParser.CMD_USER_ATTR);
+        cmdArgs.add("attr1=val1,attr3=val4");
+
+        ConnectionAndSslParameters args = parseArgs(cmdArgs);
+
+        assertEquals(args.userAttributes().get("attr3"), "val4");
```

Comment:
We are to ensure all other attributes from both sources are still here.

Reply:
done